pkg/compactor/compactor.go

```diff
@@ -581,6 +596,11 @@ func (c *Compactor) starting(ctx context.Context) error {
 }
 
 func (c *Compactor) stopping(_ error) error {
+	begin := time.Now()
+	defer func() {
+		c.CompactorStopDurationSeconds.Set(time.Since(begin).Seconds())
```

Comment:
After emitting the metric, compactor process stops immediately.
Then how can Prometheus scrapes it?

Reply:
Could we add a finalSleep similar to the lifecyler? So we add a finalSleep here to make sure the stopDuration metrics are collected.

https://github.com/cortexproject/cortex/blob/master/pkg/ring/lifecycler.go#L895

Reply:
I don't think it makes sense to add a new flag of sleep time in order to catch a metric.

Reply:
Discussed offline, using log instead of metric for stop duration

Reply:
@wenxu1024 seems you are right. The final sleep is already available in code and we just need to expose it via compactor flag.
But Idk if it is worth to wait a longer time when stopping in order to capture the metric.

Reply:
got it. Seems to collect this new metric, we need to sleep in this stopping function (not in the lifecycler). A log would be sufficient for now.